app/controllers/orders_controller.rb

```diff
@@ -0,0 +1,64 @@
+class OrdersController < ApplicationController
+  def index
+    @orders = Order.for_user(current_user)
+  end
+
+  def show
+    @order = Order.find(params[:id])
+  end
+
+  def no_cart
+    flash[:alert] = 'you have not put anything in the cart'
+    redirect_to root_path
+  end
+
+  def create
+    if current_user.present?
+      order = find_order
+      if order.blank?
+        params[:alert] = "This Order is not belongs to you"
+        redirect_to :back
+        return
+      end
+
+      if !params[:order_detail][:quantity].to_s.match(/\A[+-]?\d+?(\.\d+)?\Z/) || params[:order_detail][:quantity].to_i < 0
```

Comment:
```ruby
quantity = params[:order_detail][:quantity]
if quantity =~ /^\d+$/ && quantity.to_i > 0
   # bla bla bla
end
```

Or I think it is better:

```ruby
quantity = params[:order_detail][:quantity].to_i
if quantity > 0
  # bla bla bla
end
```

Because whatever it is, `to_i` will always convert a String to an Integer right? So checking if it contains other than number is unnecessary, because by then it will be converted to 0.